### PR TITLE
Support collection deletion & recreation

### PR DESCRIPTION
**Pull request by plmercereau**
so collection is set again when it is deleted and recreated

### CODE REVIEW

**Comment by cvara** [comment]
Thanks for this @plmercereau.

Could you also add a test case that covers collection deletion + recreation? Something along the lines of this:
https://github.com/cvara/rxdb-hooks/blob/master/tests/useRxData.test.tsx#L735

**Comment by plmercereau** [comment]
I just pushed a test, but I am not sure it is correct - I'm sorry I am don't really know how to perform tests in React...
The idea is 1. to create a collection - and to use it in the component, 2. to remove it - and to make sure RxData can't find anything and 3. to recreate it and to use the new collection in RxData again.

File: tests/useRxData.test.tsx [hunk 1]
```diff
@@ -796,4 +796,60 @@ describe('useRxData + lazy collection init', () => {
...
+					id: '1',
+					name: 'Boba Fett',
+					affiliation: 'Mandalorian',
+					age: 56,
+				},
+			];
+			await setupCollection(db, wrongCharacters, 'characters');
+			await db.removeCollection('characters');
+			await setupCollection(db, characters, 'characters');
+		});
```
Comment (cvara) [inline tests/useRxData.test.tsx:843]: This is in the right direction. 

Nitpicking really, but to make the test a bit more strict, you could split this into 2 `act()` invocations, one for initially creating the collection and another for deleting/recreating it. Each `act()` could then be followed by 2 assertions: one verifying that the docs of that particular collection are rendered and another verifying that docs of the 'wrong' collection are not.

File: src/useRxCollection.ts [hunk 2]
```diff
@@ -17,7 +17,7 @@ function useRxCollection<T>(name: string): RxCollection<T> | null {
 			const sub = db.newCollections$.subscribe(col => {
 				if (col[name]) {
 					setCollection(col[name]);
-					sub.unsubscribe();
+					// sub.unsubscribe();
```
Comment (cvara) [inline src/useRxCollection.ts:20]: You can drop the commented line altogether

**Review by cvara — CHANGES_REQUESTED** [review]

**Pull request merged**